### ISSUE TITLE
Fix #42 without disabling ghosting indicators

### ISSUE THREAD

**Issue by NanoAi** [NanoAi/gm_apg#44]
Currently when dealing with #42  this was fixed by just wrapping `SetColor` in a `if APG.cfg["ghostColorToggle"].value then` statement.

I feel like a more elegant solution would be to check when an object has `duplicator.DoGeneric` on to it, and grab the colour data from there so that the ghosting can work as intended with the intended visual indicators.

**Comment by AvarianKnight** [comment]
While I feel this would be a better solution, I think it's still a good idea to give people the option to disable the ghosting color.

**Comment by NanoAi** [comment]
Yeah the option is nice, but it might be confusing for some users. Plus it doesn't fix the actual issue, just works around it.
